lib/Navigation/Builder.php

```diff
@@ -114,38 +114,25 @@ protected function resolveOptions(array $options): array
     }
 
     /**
-     * @param array|Document|null $activeDocument
-     * @param Document|null $navigationRootDocument
-     * @param string|null $htmlMenuIdPrefix
-     * @param \Closure|null $pageCallback
-     * @param bool|string $cache
-     * @param int|null $maxDepth
-     * @param int|null $cacheLifetime
-     *
+     * @param array $params
      * @return mixed|\Pimcore\Navigation\Container
```

Comment:
Should only return `\Pimcore\Navigation\Container`, right? So we can add this as return type hint. Maybe we can optimize the part with `CacheManager::load()` from line 144

```
        $navigation = null
        if ($cacheEnabled) {
            // the cache key consists out of the ID and the class name (eg. for hardlinks) of the root document and the optional html prefix
            $cacheKeys = ['root_id__' . $navigationRootDocument->getId(), $htmlMenuIdPrefix, get_class($navigationRootDocument)];
            if (Site::isSiteRequest()) {
                $site = Site::getCurrentSite();
                $cacheKeys[] = 'site__' . $site->getId();
            }
            if (is_string($cache)) {
                $cacheKeys[] = 'custom__' . $cache;
            }
            if ($pageCallback instanceof \Closure) {
                $cacheKeys[] = 'pageCallback_' . closureHash($pageCallback);
            }
            if ($maxDepth) {
                $cacheKeys[] = 'maxDepth_' . $maxDepth;
            }
            $cacheKey = 'nav_' . md5(serialize($cacheKeys));
            $navigation = CacheManager::load($cacheKey);
        }
        if (!$navigation instanceof \Pimcore\Navigation\Container) {
```

Reply:
@blankse Thanks, you are right. It will return \Pimcore\Navigation\Container. Accepted the php Doc change.